### PR TITLE
Support URIs in part-of directives.

### LINKED ISSUE #615: Allow using URI strings in "part of" declarations
According to the 1.23 CHANGELOG this is now supported in the language, but dartfmt doesn't seem to be happy with it.

dartfmt 0.2.16 on the command line removes a "part of" URI from the file if I use one.

Formatting via the dart_style library seems to refuse to format files using a URI "part of" -- seen in this pull request for source_gen trying to change the generated code: https://github.com/dart-lang/source_gen/pull/138

### PR DESCRIPTION
**Pull request by munificent**
Fix #615.

### CODE REVIEW

File: test/splitting/function_types.unit [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+Function(parameter, {int foo, String bar}) f;
+<<<
+Function(parameter,
+    {int foo, String bar}) f;
+>>> avoid splitting in function type parameters
+Function(parameter1, void printFn(param1, param2)) f;
+<<<
+Function(parameter1,
+    void printFn(param1, param2)) f;
+>>>
```
Comment (natebosch) [inline test/splitting/function_types.unit]: [nit] no name on this one? what is interesting about having a shorter first parameter?

File: test/splitting/function_types.unit [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    parameter2,
+    parameter3]) f;
+>>> split between type and name
+Function(VerylongParameterType parameterName) f;
+<<<
+Function(
+    VerylongParameterType
+        parameterName) f;
+>>> split in function type and on variable name
+Function(VeryVeryVeryVeryLongParameterType) veryLongVariableName;
```
Comment (natebosch) [inline test/splitting/function_types.unit]: [nit] this isn't a VeryVeryVeryVeryLongParameterType it's a veryVeryVeryVeryLongParameterName

File: test/splitting/function_types.unit [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+<<<
+Function(
+        VeryVeryVeryVeryLongParameterType)
+    veryLongVariableName;
+>>> split in nested function type forces outer split
+Function(int, String, Function(parameter1, parameter2, parameter3)) f;
+<<<
+Function(
+    int,
```
Comment (natebosch) [inline test/splitting/function_types.unit]: [nit] this is misleading... IIUC this is a parameter named 'int' of type 'dynamic'. Or am I wrong?

File: test/splitting/function_types.unit (hunk 1 above)
Comment (munificent) [inline test/splitting/function_types.unit]: Yeah, not all of the tests have descriptions. I try to add them when it's worthwhile, but it's often obvious from the test.

In this case, it was an accidental copy/paste error and is redundant with the above one. Removed it.

File: test/splitting/function_types.unit (hunk 2 above)
Comment (munificent) [inline test/splitting/function_types.unit]: It is actually a type (though it doesn't matter either way for this test). Unlike the old function type syntax, the new syntax wisely assumes if you omit one identifier, you're omitting the name, not the type.

File: test/splitting/function_types.unit (hunk 3 above)
Comment (munificent) [inline test/splitting/function_types.unit]: See above.

**Pull request merged**